src/main/java/com/iexec/sms/secret/web2/Web2SecretRepository.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *
+ *  * Copyright 2022 IEXEC BLOCKCHAIN TECH
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package com.iexec.sms.secret.web2;
+
+import org.springframework.data.repository.CrudRepository;
+
+import java.util.Optional;
+
+public interface Web2SecretRepository extends CrudRepository<Web2Secret, String> {
```

Comment:
I don't know if it makes a lots of difference, I used the following inheritance in past projects
```suggestion
public interface Web2SecretRepository extends JpaRepository<Web2Secret, String> {
```

If you know if a variant is better than the other, I am interested

Reply:
Extracted from [this SO answer](https://stackoverflow.com/a/14025100/7421571):
> [JpaRepository](http://static.springsource.org/spring-data/data-jpa/docs/current/api/org/springframework/data/jpa/repository/JpaRepository.html) extends [PagingAndSortingRepository](http://static.springsource.org/spring-data/data-commons/docs/current/api/org/springframework/data/repository/PagingAndSortingRepository.html) which in turn extends [CrudRepository](http://static.springsource.org/spring-data/data-commons/docs/current/api/org/springframework/data/repository/CrudRepository.html).
>
>Their main functions are:
>
>- [CrudRepository](http://static.springsource.org/spring-data/data-commons/docs/current/api/org/springframework/data/repository/CrudRepository.html) mainly provides CRUD functions.
>- [PagingAndSortingRepository](http://static.springsource.org/spring-data/data-commons/docs/current/api/org/springframework/data/repository/PagingAndSortingRepository.html) provides methods to do pagination and sorting records.
>- [JpaRepository](http://static.springsource.org/spring-data/data-jpa/docs/current/api/org/springframework/data/jpa/repository/JpaRepository.html) provides some JPA-related methods such as flushing the persistence context and deleting records in a batch.

As we don't need any of `PagingAndSortingRepository` nor `JpaRepository` methods, I think we can keep `CrudRepository`.

Reply:
Thanks for the link, we can keep `CrudRepository`